subsys/mgmt/mcumgr/smp/src/mcumgr_client.c

```diff
@@ -0,0 +1,756 @@
+
+	ok = zcbor_map_decode_bulk(zsd, echo_response, ARRAY_SIZE(echo_response), &decoded) == 0;
+
+	if (!ok) {
+		response.status = MGMT_ERR_ECORRUPT;
+		client_gr_os_fn(&response);
+		k_sem_give(user_data);
+		return MGMT_ERR_ECORRUPT;
+	}
+
+	response.status = MGMT_ERR_EOK;
+	client_gr_os_fn(&response);
+	k_sem_give(user_data);
+	return 0;
+}
+
+static int reset_res_fn(struct net_buf *nb, void *user_data)
+{
+	struct mcumgr_client_os_gr_rsp response;
+
+	if (!nb) {
+		response.status = MGMT_ERR_ETIMEOUT;
+	} else {
+		response.status = MGMT_ERR_EOK;
+	}
+	client_gr_os_fn(&response);
+	k_sem_give(user_data);
+	return 0;
+}
+
+static void reset_image_map_decoder_list(struct zcbor_map_decode_key_val *map, size_t map_size)
+{
+	for (int i = 0; i < map_size; i++) {
+		map[i].found = false;
+		map[i].value_ptr = &image_info[image_count];
+	}
+
+	image_info[image_count].active = false;
+	image_info[image_count].bootable = false;
+	image_info[image_count].confirmed = false;
+	image_info[image_count].pending = false;
+	image_info[image_count].permanent = false;
+	image_info[image_count].img_num = 0;
+	image_info[image_count].slot_num = -1;
+
+}
+
+static bool get_image_version(zcbor_state_t *state, struct mcumgr_image_data *out)
+{
+	struct zcbor_string val = {0};
+	bool fin = zcbor_tstr_decode(state, &val);
+
+	if (!fin || !val.len) {
+		return false;
+	}
+
+	if (val.len >= sizeof(out->version)) {
+		val.len = sizeof(out->version) - 1;
+	}
+	memcpy(out->version, val.value, val.len);
+	out->version[val.len] = '\0';
+	return true;
+}
+
+static bool get_image_hash(zcbor_state_t *state, struct mcumgr_image_data *out)
+{
+	struct zcbor_string val = {0};
+	bool fin = zcbor_bstr_decode(state, &val);
+
+	if (!fin || !val.len) {
+		return false;
+	}
+
+	if (val.len >= sizeof(out->hash)) {
+		val.len = sizeof(out->hash) - 1;
+	}
+	memcpy(out->hash, val.value, val.len);
+	out->hash[val.len] = '\0';
+	return true;
+}
+
+static bool get_image_slot(zcbor_state_t *state, struct mcumgr_image_data *out)
+{
+	bool fin = zcbor_int32_decode(state, &out->slot_num);
+
+	return fin;
+}
+
+static bool get_image_num(zcbor_state_t *state, struct mcumgr_image_data *out)
+{
+	bool fin = zcbor_int32_decode(state, &out->img_num);
+
+	return fin;
+}
+
+static bool get_image_bootable(zcbor_state_t *state, struct mcumgr_image_data *out)
+{
+	bool fin = zcbor_bool_decode(state, &out->bootable);
+
+	return fin;
+}
+
+static bool get_image_pending(zcbor_state_t *state, struct mcumgr_image_data *out)
+{
+	bool fin = zcbor_bool_decode(state, &out->pending);
+
+	return fin;
+}
+
+static bool get_image_confirmed(zcbor_state_t *state, struct mcumgr_image_data *out)
+{
+	bool fin = zcbor_bool_decode(state, &out->confirmed);
+
+	return fin;
+}
+
+static bool get_image_active(zcbor_state_t *state, struct mcumgr_image_data *out)
+{
+	bool fin = zcbor_bool_decode(state, &out->active);
+
+	return fin;
+}
+
+static bool get_image_pernament(zcbor_state_t *state, struct mcumgr_image_data *out)
```

Comment:
It should be storing the data *locally* so it can be verified before putting it in a global or shared scope - see https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/mgmt/mcumgr/grp/img_mgmt/src/img_mgmt.c#L391 it should be working similarly to that

Reply:
OK. let me put back those local buffer's and copy those to image data list.